In [4]:
import sys
print(sys.executable)

c:\Users\harshitha\Projects\flipkart_capstone\.venv\Scripts\python.exe


In [5]:
import pandas as pd

df = pd.read_csv("../data/orders_dataset.csv")

# 1. Total row count
total_rows = len(df)
print(f"Total rows: {total_rows}")

# 2. Overall return rate
return_rate = df['returned'].mean()
print(f"Overall return rate: {return_rate:.4f} ({return_rate*100:.2f}%)")

# 3. Missing rating_given
missing_pct = df['rating_given'].isna().mean()
print(f"Missing rating_given: {missing_pct:.4f} ({missing_pct*100:.2f}%)")

# 4. Return rate by product_category
print("\nReturn rate by product_category:")
print(df.groupby('product_category')['returned'].mean().sort_values(ascending=False))

# 5. Return rate by payment_method
print("\nReturn rate by payment_method:")
print(df.groupby('payment_method')['returned'].mean().sort_values(ascending=False))

# 6. Missingness of rating_given by product_category and payment_method
categorical_cols = ['product_category', 'payment_method']

for col in categorical_cols:
    print(f"\nMissing rating_given rate by {col}:")
    print(df.groupby(col)['rating_given'].apply(lambda x: x.isna().mean()).sort_values(ascending=False))

# 7. Missingness of rating_given by payment_method (the MAR evidence)
print("\nMissing rating_given rate by payment_method:")
print(df.groupby('payment_method')['rating_given'].apply(lambda x: x.isna().mean()).sort_values(ascending=False))

Total rows: 6000
Overall return rate: 0.2275 (22.75%)
Missing rating_given: 0.1305 (13.05%)

Return rate by product_category:
product_category
Apparel        0.264275
Footwear       0.259570
Beauty         0.200345
Home           0.191469
Electronics    0.186930
Name: returned, dtype: float64

Return rate by payment_method:
payment_method
COD             0.307477
Wallet          0.178451
Prepaid_UPI     0.169199
Prepaid_Card    0.168154
Name: returned, dtype: float64

Missing rating_given rate by product_category:
product_category
Home           0.152607
Electronics    0.139058
Footwear       0.130719
Beauty         0.129534
Apparel        0.113188
Name: rating_given, dtype: float64

Missing rating_given rate by payment_method:
payment_method
COD             0.228309
Wallet          0.063973
Prepaid_Card    0.063143
Prepaid_UPI     0.056630
Name: rating_given, dtype: float64

Missing rating_given rate by payment_method:
payment_method
COD             0.228309
Wallet          0.063973
P

About 13 % of 'rating given' values were missing.While comparing the missing rate across product category and payment method, the missingness in rating_given is not random but explained by another observed column, making it Missing at random, explained by payment method, where COD customers pay on delivery and often skip the rating step after delivery of product